src/app/users/postManageConsoleRoles.js

```diff
@@ -0,0 +1,62 @@
+const config = require('../../infrastructure/config');
+const { sendResult } = require('../../infrastructure/utils');
+const { getUserDetails} = require('./utils');
+const { getServiceById } = require('../../infrastructure/applications')
+const { listRolesOfService, updateUserService } = require('../../infrastructure/access');
+const { getSingleServiceForUser, addOrChangeManageConsoleServiceTitle, checkIfRolesChanged }= require('./getManageConsoleRoles')
+
+const postManageConsoleRoles = async (req, res) => {
+
+    let rolesSelectedNew = req.body.role ? req.body.role : [];
+    if (!(rolesSelectedNew instanceof Array)) {
+      rolesSelectedNew = [req.body.role];
+    }
+    
+    const manage = await getServiceById('manage');
+    const serviceSelectedByUser = await getServiceById(req.params.sid);
+    const user = await getUserDetails(req);
+    const userManageRoles = await getSingleServiceForUser(req.params.uid, config.access.identifiers.departmentForEducation, manage.id, req.id);
+    const manageConsoleRolesForAllServices = await listRolesOfService(manage.id);
+    const manageConsoleRolesForSelectedService = manageConsoleRolesForAllServices.filter(service => service.code.split('_')[0] === req.params.sid);
+    let manageConsoleRoleIds = [];
+    manageConsoleRolesForSelectedService.forEach(obj => manageConsoleRoleIds.push(obj.id));
+    
+    const addOrChangeService = addOrChangeManageConsoleServiceTitle(userManageRoles, manageConsoleRoleIds);
+    
+    for (let i =0; i < rolesSelectedNew.length; i++) {
+      if(!manageConsoleRoleIds.includes(rolesSelectedNew[i])) {
+        sendResult(req, res, 'users/views/selectManageConsoleRoles', {
+          csrfToken: req.csrfToken(),
+          layout: 'sharedViews/layoutNew.ejs', 
+          user,
+          serviceSelectedByUser,
+          addOrChangeService,
+          manageConsoleRolesForSelectedService,
+          userManageRoles,
+          backLink: true,
+          validationMessages: {
+            roleSelection: 'You have selected an invalid role for this service'
+          },
+          cancelLink: `/users/${user.id}/organisations`,
+        });
+      } 
+    }
+    
+    let currentRoles = [];
+    userManageRoles.roles.forEach(role => currentRoles.push(role.id));
+
+    const allSelectedRoles = [...new Set(currentRoles.concat(rolesSelectedNew))];
+    const rolesToRemove = manageConsoleRoleIds.filter(id => !rolesSelectedNew.includes(id));
+    const newRoles = allSelectedRoles.filter(id => !rolesToRemove.includes(id));
+
+    if (!checkIfRolesChanged(currentRoles, newRoles)) {
+      return res.redirect(`/users/${req.params.uid}/manage-console-services`);
+    } else {
+      updateUserService(req.params.uid, manage.id, config.access.identifiers.departmentForEducation, newRoles, req.id);

```

Comment:
🟥 Will this work if the user doesn't already have manage console access? [Update user service](https://github.com/DFE-Digital/login.dfe.access/blob/d1be2d5520982c8cba7f2ed2df2a5adf3b16653d/src/app/users/updateUserService.js#L62-L65) appears to return 404 if the user doesn't have it already 😄 

I can see we've made the change to only show these pages if a user already has manage access, but that'll mean a script will need to be run to set up that access initially 👍

Reply:
Hey Jared, thanks for the feedback - I did think the update would add the service if the user didnt have it, but thanks for pointing out that this isnt the case. I will add it in explicitly 😊